subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/inputs/undeclared/UndeclaredBuildInputsIntegrationTest.groovy

```diff
@@ -403,6 +403,29 @@ class UndeclaredBuildInputsIntegrationTest extends AbstractConfigurationCacheInt
         EnvVariableRead.getEnvContainsKey("CI")             | "false"    | "defined" | "true"
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/19710")
+    def "modification of allowed properties do not invalidate cache"() {
+        buildFile("""
+            System.setProperty("java.awt.headless", "true")
+
+            // Attempt to capture the modified property value.
+            println("java.awt.headless=\${System.getProperty("java.awt.headless")}")
```

Comment:
⭕ I'd move the printing to an ad-hoc task so its output can be asserted on each run. Asserting on the output would serve to document the expected behavior more explicitly (it would be 'true' on the first run and 'null' on the second?)

Reply:
I'm not a big fan of asserting incorrect behavior but I don't have strong objections, so I've added the printing task. However, the property has to be captured at the configuration time too, this is the very essence of the reproducer.

> it would be 'true' on the first run and 'null' on the second?

Yes, `null` is consistent with what we've had at 7.3 but incorrect, of course.

Reply:
I understand the reluctance but in my opinion is better than to leave the reader wondering what the behavior would be in this case. Thanks!